test/test-parse-function.cpp

```diff
@@ -1658,6 +1658,16 @@ TEST_F(test_parse_function, function_as_with_statement_body_is_disallowed) {
   }
 }
 
+TEST_F(test_parse_function, function_as_label_body_is_allowed) {
+  test_parser p(u8"l: function f() {}"_sv, capture_diags);
+  p.parse_and_visit_statement();
+  EXPECT_THAT(p.visits, ElementsAre("visit_variable_declaration",
+                                    "visit_enter_function_scope",
+                                    "visit_enter_function_scope_body",
+                                    "visit_exit_function_scope"));
+  EXPECT_THAT(p.errors, IsEmpty());
```

Comment:
Nit: Omit `capture_diags` if you're asserting no errros:
```suggestion
  test_parser p(u8"l: function f() {}"_sv);
  p.parse_and_visit_statement();
  EXPECT_THAT(p.visits, ElementsAre("visit_variable_declaration",
                                    "visit_enter_function_scope",
                                    "visit_enter_function_scope_body",
                                    "visit_exit_function_scope"));
```

Reply:
Fixed! I actually had a question about this `capture_diags` too: I don't think I fully understand what it does and how it works. I looked at the code at `parse-support.h` and `parse-support.cpp` and realized it is present in two constructors, but I don't see it being actually used

Reply:
`capture_diags` is a tag without any state. If `capture_diags` is *not* set, reporting any diagnostic will fail the test. If `capture_diags` is set, diagnostics will be saved in the `p.errors` vector, so you should check them later with `EXPECT_THAT(p.errors, ...)`.

Reply:
Oh, so I don't even need to assert that `p.errors` is empty, that's very neat!

I have changed the code to remove the assertions in these cases